tests/regression/test_voting_acc.py

```diff
@@ -121,3 +169,147 @@ def test_can_object(stranger, test_vote):
+    assert list(vote_for_tx.events["AttemptCastVoteAsDelegate"]["voters"]) == voters
+    assert contracts.voting.getVoterState(vote_id, voters[0]) == VoterState.Absent.value
+    assert contracts.voting.getVoterState(vote_id, voters[1]) == VoterState.Absent.value
+    assert contracts.voting.getVoterState(vote_id, voters[2]) == VoterState.DelegateYea.value
+    assert contracts.voting.getVote(vote_id)["yea"] == contracts.ldo_token.balanceOf(voters[2])
+
+    # A delegate can change their mind and vote for the opposite
+    vote_for_tx = contracts.voting.attemptVoteFor(vote_id, False, voters[2], {"from": delegate1})
+    # Check events and state
+    assert contracts.voting.getVoterState(vote_id, voters[2]) == VoterState.DelegateNay.value
+    assert contracts.voting.getVote(vote_id)["nay"] == contracts.ldo_token.balanceOf(voters[2])
+
+    # A voter can overwrite the delegate vote
+    contracts.voting.vote(vote_id, True, False, {"from": accounts.at(voters[2], force=True)})
+    assert contracts.voting.getVoterState(vote_id, voters[2]) == VoterState.Yea.value
+    assert contracts.voting.getVote(vote_id)["yea"] == contracts.ldo_token.balanceOf(voters[2])
+    assert contracts.voting.getVote(vote_id)["nay"] == 0
+    # A delegate can not overwrite the vote of the voter
+    with reverts("VOTING_CAN_NOT_VOTE_FOR"):
+        contracts.voting.attemptVoteFor(vote_id, True, voters[2], {"from": delegate1})
+
+    # Prepare state for the next test
+    chain.snapshot()
+    contracts.voting.attemptVoteFor(vote_id, True, voters[1], {"from": delegate2})
+    contracts.voting.assignDelegate(delegate1, {"from": accounts.at(voters[0], force=True)})
+    contracts.voting.attemptVoteFor(vote_id, True, voters[0], {"from": delegate1})
+    # Fast-forward to the objection phase
+    chain.sleep(contracts.voting.voteTime() - contracts.voting.objectionPhaseTime())
+    chain.mine()
+    assert contracts.voting.getVotePhase(vote_id) == 1  # Objection phase
+
+    # A delegate can not vote for a voter that overwrote their vote during the objection phase
+    with reverts("VOTING_CAN_NOT_VOTE_FOR"):
+        contracts.voting.attemptVoteFor(vote_id, False, voters[2], {"from": delegate1})
+    # A voter can overwrite the delegate vote during the objection phase
+    contracts.voting.vote(vote_id, False, False, {"from": accounts.at(voters[1], force=True)})
+    assert contracts.voting.getVoterState(vote_id, voters[1]) == VoterState.Nay.value
+    # A delegate can vote for a voter during the objection phase
+    contracts.voting.attemptVoteFor(vote_id, False, voters[0], {"from": delegate1})
+
+    # A vote can be safely executed
+    chain.revert()
+    # Add some more VP to be able to execute the vote
+    contracts.voting.attemptVoteFor(vote_id, True, voters[1], {"from": delegate2})
+    # Fast-forward to the closed phase
+    chain.sleep(contracts.voting.voteTime())
+    chain.mine()
+    assert contracts.voting.getVotePhase(vote_id) == 2  # Closed phase
```

Comment:
It would also be nice to assert that the voting reached a quorum before execution (the total voting power is enough to count it passed).

Reply:
Fixed in 98c41312c4dcbf45286acc799ee16c3e46be578a.